### PR TITLE
[feat] Implement SHRED marker

### LINKED ISSUE #39: It should be possible to remove an entire file
I have two ideas off the top of my head:

1. Have a file with list that specifies files to be removed (which always includes the file with the list itself)
2. OR, have some other marker that's placed at the top of a file, e.g. `REPOBEE-SANITIZER-SHRED`

I think the second option seems more smooth, and it's probably easier.

### LINKED ISSUE #73: Add the remove file marker to documentation
After #39 is finished, reflect it in the documentation.

This includes the README, and the syntax module

### PR DESCRIPTION
**Pull request by tohanss**
If a file now contains a `REPOBEE-SANITIZER-SHRED` marker, the entire file will be removed from the file system.

This is possible using the `Sanitize-file` command but is most useful when working with repositories.

Fix #39 and fix #73

### CODE REVIEW

File: repobee_sanitizer/_syntax.py [hunk 1]
```diff
@@ -66,6 +66,9 @@ def check_syntax(lines: List[str]) -> None:
     prefix = ""
     has_blocks = False
 
+    if _contained_marker(lines[0]) == Markers.SHRED:
+        has_blocks = True
+
```
Comment (tohanss) [inline repobee_sanitizer/_syntax.py:71]: This is the fix to the bug I mentioned me i figured out as i was entering my sleep

File: repobee_sanitizer/_sanitize.py [hunk 2]
```diff
@@ -3,19 +3,42 @@
...
 from repobee_sanitizer import _syntax
 
 import re
-from typing import List, Iterable
+from typing import List, Iterable, Optional
 
 
-def sanitize(content: str) -> str:
-    """Create a sanitized version of the input.
+def sanitize_file(file_abs_path: pathlib.Path) -> Optional[str]:
```
Comment (slarse) [inline repobee_sanitizer/_sanitize.py:14]: Good type hinting.

File: tests/test_ext_commands.py [hunk 3]
```diff
@@ -113,6 +113,33 @@ def test_sanitize_repo_discover_mode_target_branch_with_binary_files(
...
+    target_branch = "student-version"
+    repobee.main(
+        f"repobee --config-file {sanitizer_config} sanitize-repo "
+        f"--target-branch {target_branch} "
+        f"--discover-files "
+        f"--repo-root {fake_repo.path}".split()
+    )
+
+    fake_repo.repo.git.checkout(target_branch)
+    fake_repo.repo.git.reset("--hard")
```
[the remark this answers is not in the document]
Comment (tohanss) [reply tests/test_ext_commands.py:139]: I'll open an issue for this, i think its a good idea, and i would make it as a separate PR

File: repobee_sanitizer/_syntax.py [hunk 4]
```diff
@@ -126,3 +134,37 @@ def file_is_dirty(
...
+    """Tells us whether or not the text has valid usage of the shred marker.
+    Valid syntax is, if a shred marker is on the first line of text then on
+    every line other than the first, there should be no markers. If there is
+    is no shred marker on the first line, then there should not be one later.
+
+    Args:
+        lines: The file to check syntax for as a list of one string per line.
+        errors: The list to send any found errors.
+    """
+    errors = []
```
[the remark this answers is not in the document]
Comment (tohanss) [reply repobee_sanitizer/_syntax.py:149]: I understand this but i'm not sure if youre asking me to change anything

**Comment by slarse** [comment]
@tohanss You still must do this:

> Pin the version number of repobee (i.e. specify it with == instead of >=) to 3.0.0-alpha.1 in setup.py and requirements/app.txt. Otherwise you will get flaky behavior in CI as 3.0.0-alpha.2 has breaking changes (see #72 )

**Comment by tohanss** [comment]
@slarse Check out #76

**Comment by tohanss** [comment]
The last commit might give us a merge conflict but I can deal with that

**Comment by slarse** [comment]
@tohanss You pinned to the wrong version, you must pin to 3.0.0-alpha.1. alpha.2 is what you're fixing support for in another #76

File: README.md [hunk 5]
```diff
@@ -149,6 +149,9 @@ for `repobee-sanitizer` to work, marker syntax must be correct, this includes sp
 - REPOBEE-SANITIZER-END
     - REQUIRED: Must exist for each start block
     - Indicates the end of a block.
+- REPOBEE_SANITIZER-SHRED
+    - OPTIONAL: If this exists, there cannot be any other markers of any type in the file
+    - Having this marker will remove the entire file when running the `sanitize-repo` or `sanitize-file` commands
```
Comment (slarse) [inline README.md:154]: Should say that the shred marker must appear on the first line

File: repobee_sanitizer/_syntax.py [hunk 6]
```diff
@@ -33,7 +34,8 @@ def check_syntax(lines: List[str]) -> None:
     .. code-block:: raw
 
         FILE ::=
-            MARKERLESS_LINE* ((BLOCK | PREFIXED_BLOCK) MARKERLESS_LINE*)+
+            SHRED_MARKER MARKERLESS_LINE |
```
Comment (slarse) [inline repobee_sanitizer/_syntax.py:37]: bug in the grammar

```suggestion
            SHRED_MARKER MARKERLESS_LINE* |
```

File: repobee_sanitizer/_syntax.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (slarse) [reply repobee_sanitizer/_syntax.py:150]: This was supposed to be a comment on one of my code snippets. Not sure where that buggered off to.

**Pull request merged**